pkg/object/mqttproxy/session.go

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package mqttproxy
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/eclipse/paho.mqtt.golang/packets"
+)
+
+type (
+	SessionManager struct {
+		smap sync.Map
```

Comment:
Why do you store sessions in the local? why don't you save it in the etcd store?

Reply:
Actually, it should be stored in EG's embedded Etcd. But we haven't implemented it in phase one. I think @suchen-sci is working on add this durable session handling and connection resuing in phase two currently.

Reply:
Yes, I am working on it. I have just finished the subscribe part, and now in phase two, only left http endpoint and session persistency to finish.

Reply:
Not put session in etcd at the beginning is mainly for convenience of testing, I actually write about 500 lines of test code to test the correctness of code.